File: point_dist.m
mat = load('./data/positions.xyz');
mat = sortrows(mat,-1);
n_lines = size(mat,1);

count = 0;
for ix = 1:n_lines
	for jx = ix+1:n_lines
		if mat(ix,1)-mat(jx,1) > 0.05
			break;
		end
			if norm(mat(ix,:)-mat(jx,:)) < 0.05
			count = count+1;
		end
	end
end

fprintf("close points: %d\n", count);

File: point_dist.py
import numpy as np

if __name__=="__main__":
    mat = np.loadtxt("./data/positions.xyz")
    mat = mat[mat[:, 0].argsort()[::-1]]
    n_lines = mat.size//3

    count = 0
    for ix in range(n_lines):
        for jx in range(ix+1, n_lines):
            if mat[ix,0]-mat[jx,0] > 0.05:
                break
            if np.linalg.norm(mat[ix]-mat[jx]) < 0.05:
                count+=1
    print("close points:", count)
